### PR TITLE
Imaging Pipeline Update

### PR DESCRIPTION
**Pull request by walyssonBarbosa**
Here is what I am adding to/changing in the pipeline:
- Updated setup_logging.py with new log information
- Added psutil module to setup.py
- Changed num_cores value on cfg files
- Added number of cores to settings.yaml and template_settings.yaml files
- Created a script to check analysis performance in different environments
- Changed a few things in imaging_pipeline.py and run_imaging_pipeline.py

### CODE REVIEW

**Comment by acviana** [comment]
Please remove the two `astrodrizzle_cfg/z3_neptune_*.cfg` files, we no longer need them. Instead the pipeline now uses the `wfpc2_centerslice.cfg` files you created.

File: scripts/production/run_imaging_pipeline.py [hunk 1]
```diff
@@ -87,29 +95,18 @@ def run_imaging_pipeline():
...
+    logging.info("Command-line arguments used:")
     for arg in args.__dict__:
-        logger.info(arg + ": " + str(args.__dict__[arg]))
+        logging.info(arg + ": " + str(args.__dict__[arg]))
     rootfile_list = glob.glob(args.filelist)
-    rootfile_list = [x for x in rootfile_list if len(os.path.basename(x)) == 18]
+    rootfile_list = [filename for filename
+                     in rootfile_list
+                     if len(filename.split('/')[-1]) == 18
+                     and filename.split('/')[-1].split('_')[-1] == 'c0m.fits']
```
Comment (acviana) [inline scripts/production/run_imaging_pipeline.py:105]: So I'm wondering why this was changed. If the basename of a file is only 18 characters long, why do we need to check to see if it's a `c0m.fits` file? I think the only files that can be that short are the `c0m.fits` files right?

File: scripts/checks/check_filesystem_completeness.py [hunk 2]
```diff
@@ -59,11 +59,11 @@ def check_filesystem_completeness_main():
     found = 0
     for filename in c0m_file_list:
         proposal_folder = filename.split('/')[-2]
-        if proposal_folder not in check_dict.keys():
+        if proposal_folder not in check_dict:
             check_dict[proposal_folder] = defaultdict(int)
         check_dict[proposal_folder]['input_file'] += 1
         file_dict = make_output_file_dict(filename)
-        for key in file_dict.iterkeys():
+        for key in file_dict.keys():
```
Comment (acviana) [inline scripts/checks/check_filesystem_completeness.py]: Why was this changed from `.iterkeys()` to `.keys()`? `.iterkeys()` is faster because it's a generator.

File: scripts/production/run_imaging_pipeline.py (hunk 1 above)
Comment (walyssonBarbosa) [inline scripts/production/run_imaging_pipeline.py:105]: Once I found a file that ended in `drz.fits` that had the same length of a `c0m.fits` file. That's why I changed that.

File: scripts/checks/check_filesystem_completeness.py (hunk 2 above)
Comment (walyssonBarbosa) [inline scripts/checks/check_filesystem_completeness.py]: Yes, it was a mistake. I am going to change that.

**Comment by acviana** [comment]
OK. I am accepting this merge. Thanks for your hard work.

:+1:
